### DISCUSSION TITLE
System.InvalidOperationException: 'XML expected.'

### DISCUSSION THREAD

**Discussion by cptb** [bruce-dunwiddie/tsql-parser#126] [General]
Hi Bruce,
It's been a while that we discussed issues. Here is another one to bring headaches :-)
We are looking at parsing this piece of SQL below
`INSERT INTO table_a(c1, c2, c3, c4)
SELECT 
    22, 
    NEXT VALUE FOR my_sequence_object, 
    160, 
    45;`

I think it does not recognise the NEXT VALUE FOR syntax. Once it appears, this line of code 
`var statements = TSQLStatementReader.ParseStatements(sql);`
throws System.InvalidOperationException: 'XML expected.'

Could you please check this? Appreciate your help.

Cheers
Paul

**Comment by bruce-dunwiddie** [comment]
As always, thanks for letting me know. I'll take a look.

**Comment by bruce-dunwiddie** [comment]
It appears that I had created a placeholder issue a while back to verify parsing functionality for sequences, https://github.com/bruce-dunwiddie/tsql-parser/issues/90 , but hadn't done work towards it yet.

It appears that the answer to my placeholder validation question is no, it does not work.

I have some hopes that it will be a fairly simple fix, but there are a few other bugs that have recently popped up that should probably all get resolved for the next release.

**Reply by cptb** [reply]
Take your time Bruce. We have parsed thousands of queries so far. This is the very first one we saw using NEXT VALUE FOR syntax, proving it's not a popular statement in our collection. Looking forward to the next release.

**Comment by SalamiArmy** [comment]
I've encountered this too. Switching to ScriptDOM until the next release.
